### PR TITLE
FAT-42626: create one activity  3 fragments

### PR DESCRIPTION
**Pull request by chikebaba**
first is 2 textview
second is echo plainText
third is recycler view with items
and drawerlayout with menu

### CODE REVIEW

File: .idea/gradle.xml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (daddymercury) [inline .idea/gradle.xml]: This type of file should not be included into GIT changes, please delete this file from Merge request Also please add into git ignore whole .idea directory

File: .idea/misc.xml [hunk 2]
```diff
@@ -1,5 +1,28 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
```
Comment (daddymercury) [inline .idea/misc.xml]: This type of file should not be included into GIT changes, please delete this file from Merge request Also please add into git ignore whole .idea directory

File: .idea/vcs.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (daddymercury) [inline .idea/vcs.xml]: This type of file should not be included into GIT changes, please delete this file from Merge request Also please add into git ignore whole .idea directory

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -12,6 +12,18 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/Theme.Androidtask"
-        tools:targetApi="31" />
+        tools:targetApi="31">
```
Comment (daddymercury) [inline app/src/main/AndroidManifest.xml]: In your Gradle file you have **targetSdk 33** please use the same here

File: app/src/main/java/com/example/android_task/CustomAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    }
+    @NonNull
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
+        val itemView = LayoutInflater.from(parent.context)
+            .inflate(R.layout.item, parent, false)
+        return MyViewHolder(itemView)
+    }
+
+
+
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/CustomAdapter.kt]:
```suggestion
```

Comment (daddymercury) [reply app/src/main/java/com/example/android_task/CustomAdapter.kt]: Delete unnecessary  indents in the code

File: app/src/main/java/com/example/android_task/CustomAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                setTitle("Picked item")
+                setMessage("Item " + itemsList[position] )
+                //  setPositiveButton("OK", DialogInterface.OnClickListener(function = positiveButtonClick))
+                //  setNegativeButton(android.R.string.no, negativeButtonClick)
+                //   setNeutralButton("Maybe", neutralButtonClick)
+                show()
+            }
+
+
+        } // click event
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/CustomAdapter.kt]:
```suggestion
        holder.itemTextView.setOnClickListener {  
            val builder = AlertDialog.Builder(it.context)
            with(builder)
            {
                setTitle("Picked item")
                setMessage("Item " + itemsList[position] )
                show()
            }
        }
```

Comment (daddymercury) [reply app/src/main/java/com/example/android_task/CustomAdapter.kt]: delete unnesesary commented code also please pay attention to white space and indents in the code

File: app/src/main/java/com/example/android_task/CustomAdapter.kt [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+
+
+    }
+
+
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/CustomAdapter.kt]: delete these 2 lines

File: app/src/main/java/com/example/android_task/CustomAdapter.kt [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+    override fun getItemCount(): Int {
+        return itemsList.size
+    }
+
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/CustomAdapter.kt]: delete this line

File: app/src/main/java/com/example/android_task/MainActivity.kt [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        setSupportActionBar(binding.appBarMain.toolbar)
+
+       /* binding.appBarMain.fab.setOnClickListener { view ->
+            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
+                .setAction("Action", null).show()
+        }*/
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/MainActivity.kt]:
```suggestion 
```

File: app/src/main/java/com/example/android_task/MainActivity.kt [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+        // Passing each menu ID as a set of Ids because each
+        // menu should be considered as top level destinations.
+        appBarConfiguration = AppBarConfiguration(
+            setOf(
+                R.id.nav_home, R.id.nav_gallery, R.id.nav_slideshow
+            ), drawerLayout
+        )
+        setupActionBarWithNavController(navController, appBarConfiguration)
+        navView.setupWithNavController(navController)
+
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/MainActivity.kt:36]: Delete EMPTY lines

File: app/src/main/java/com/example/android_task/ui/gallery/GalleryFragment.kt [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+    galleryViewModel.text.observe(viewLifecycleOwner) {
+      textView.text = it
+    }
+    return root
+  }
+
+
+    override fun onStart() {
+        super.onStart()
+        // Toast.makeText(activity, "fragments onStart method", Toast.LENGTH_LONG).show()
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/ui/gallery/GalleryFragment.kt]: Delete all commented code!

File: app/src/main/java/com/example/android_task/ui/gallery/GalleryFragment.kt [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+            override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            }
+
+            override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
+            }
+
+        })*/
+
+
+        watchedText.addTextChangedListener(object : TextWatcher {
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/ui/gallery/GalleryFragment.kt:41]: user instead Kotlin do **_onTextChanged_** and delete Textwatcher if you don't use these 3 methods in 1 time

File: app/src/main/java/com/example/android_task/ui/gallery/GalleryFragment.kt [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+
+            override fun beforeTextChanged(s: CharSequence, start: Int,
+                                           count: Int, after: Int) {
+            }
+
+            override fun onTextChanged(s: CharSequence, start: Int,
+                                       before: Int, count: Int) {
+                Toast.makeText(context, s , Toast.LENGTH_SHORT).show()
+
+                // tvSample.setText("Text in EditText : "+s)
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/ui/gallery/GalleryFragment.kt]: commented code should be deleted!

File: app/src/main/java/com/example/android_task/ui/gallery/GalleryFragment.kt [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+                // textView4.setText("Text in EditText : "+s)
+
+            }
+        })
+
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/ui/gallery/GalleryFragment.kt]: empty lines

File: app/src/main/java/com/example/android_task/ui/slideshow/SlideshowFragment.kt [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+import com.example.android_task.CustomAdapter
+import com.example.android_task.R
+import com.example.android_task.databinding.FragmentHomeBinding
+import kotlinx.android.synthetic.main.item.*
+import kotlinx.android.synthetic.main.fragment_slideshow.*
+
+
+class SlideshowFragment : Fragment() {
+
+
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/ui/slideshow/SlideshowFragment.kt]: empty lines

File: app/src/main/java/com/example/android_task/ui/slideshow/SlideshowFragment.kt [hunk 16]
```diff
@@ -0,0 +1,110 @@
...
+
+
+    //
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/ui/slideshow/SlideshowFragment.kt]: delete it **//**

File: app/src/main/java/com/example/android_task/ui/slideshow/SlideshowFragment.kt [hunk 17]
```diff
@@ -0,0 +1,110 @@
...
+
+    private val itemsList = ArrayList<String>()
+    private lateinit var customAdapter: CustomAdapter
+
+    //
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/ui/slideshow/SlideshowFragment.kt]: delete it **//**

File: app/src/main/java/com/example/android_task/ui/slideshow/SlideshowFragment.kt [hunk 18]
```diff
@@ -0,0 +1,110 @@
...
+        val recyclerView = recyclerView as RecyclerView
+        customAdapter = CustomAdapter(itemsList)
+        val layoutManager = LinearLayoutManager(context)
+        recyclerView.layoutManager = layoutManager
+        recyclerView.adapter = customAdapter
+        prepareItems()
+    }
+
+
+    //
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/ui/slideshow/SlideshowFragment.kt]: delete it **//**

File: app/src/main/java/com/example/android_task/ui/slideshow/SlideshowFragment.kt [hunk 19]
```diff
@@ -0,0 +1,110 @@
...
+        itemsList.add("Item 5")
+        itemsList.add("Item 6")
+        itemsList.add("Item 7")
+        itemsList.add("Item 8")
+        itemsList.add("Item 9")
+        itemsList.add("Item 10")
+
+        customAdapter.notifyDataSetChanged()
+    }
+    //
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/ui/slideshow/SlideshowFragment.kt]: delete it **//**

File: app/src/main/java/com/example/android_task/ui/slideshow/SlideshowFragment.kt [hunk 20]
```diff
@@ -0,0 +1,110 @@
...
+    savedInstanceState: Bundle?
+  ): View {
+    val slideshowViewModel =
+            ViewModelProvider(this).get(SlideshowViewModel::class.java)
+
+    _binding = FragmentSlideshowBinding.inflate(inflater, container, false)
+    val root: View = binding.root
+
+    val textView: TextView = binding.textSlideshow
+    slideshowViewModel.text.observe(viewLifecycleOwner) {
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/ui/slideshow/SlideshowFragment.kt]: delete commented code

File: app/src/main/java/com/example/android_task/ui/slideshow/SlideshowFragment.kt [hunk 21]
```diff
@@ -0,0 +1,110 @@
...
+
+
+    return root
+  }
+
+
+override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/ui/slideshow/SlideshowFragment.kt]: Are you sure you need onDestroyView be commented?

File: app/src/main/java/com/example/android_task/ui/slideshow/SlideshowFragment.kt [hunk 22]
```diff
@@ -0,0 +1,110 @@
...
+
+
+    */
+
+
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/ui/slideshow/SlideshowFragment.kt]: empty lines

File: app/src/main/res/layout-land/fragment_slideshow.xml [hunk 23]
```diff
@@ -0,0 +1,37 @@
...
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".ui.slideshow.SlideshowFragment">
+
+    <TextView
+        android:id="@+id/text_slideshow"
+        android:layout_width="match_parent"
```
Comment (daddymercury) [inline app/src/main/res/layout-land/fragment_slideshow.xml:11]: instead match_parent please use 0 dp

File: app/src/main/res/layout-land/fragment_slideshow.xml [hunk 24]
```diff
@@ -0,0 +1,37 @@
...
+        android:layout_height="wrap_content"
+        android:layout_marginStart="8dp"
+        android:layout_marginTop="8dp"
+        android:layout_marginEnd="8dp"
+        android:textAlignment="center"
+        android:textSize="20sp"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="1.0"
```
Comment (daddymercury) [inline app/src/main/res/layout-land/fragment_slideshow.xml]: delete using bias

File: app/src/main/res/layout-land/fragment_slideshow.xml [hunk 25]
```diff
@@ -0,0 +1,37 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintVertical_bias="0.011" />
```
Comment (daddymercury) [inline app/src/main/res/layout-land/fragment_slideshow.xml]: delete using bias

File: app/src/main/res/layout-land/fragment_slideshow.xml [hunk 26]
```diff
@@ -0,0 +1,37 @@
...
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/recyclerView"
+        android:layout_width="594dp"
```
Comment (daddymercury) [inline app/src/main/res/layout-land/fragment_slideshow.xml]: no hardcoded allowed use constraint instead!

File: app/src/main/res/layout-land/fragment_slideshow.xml [hunk 27]
```diff
@@ -0,0 +1,37 @@
...
+        android:layout_height="227dp"
```
Comment (daddymercury) [inline app/src/main/res/layout-land/fragment_slideshow.xml]: no hard code

File: app/src/main/res/layout-land/fragment_slideshow.xml [hunk 28]
```diff
@@ -0,0 +1,37 @@
...
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.494"
```
Comment (daddymercury) [inline app/src/main/res/layout-land/fragment_slideshow.xml]: delete using bias

File: app/src/main/res/layout-land/fragment_slideshow.xml [hunk 29]
```diff
@@ -0,0 +1,37 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintVertical_bias="0.586" />
```
Comment (daddymercury) [inline app/src/main/res/layout-land/fragment_slideshow.xml]: delete using bias

File: app/src/main/res/layout-land/fragment_slideshow.xml [hunk 30]
```diff
@@ -0,0 +1,37 @@
...
+
```
Comment (daddymercury) [inline app/src/main/res/layout-land/fragment_slideshow.xml:33]: delete empty line

File: app/src/main/res/layout/app_bar_main.xml [hunk 31]
```diff
@@ -0,0 +1,34 @@
...
+            android:layout_width="match_parent"
+            android:layout_height="?attr/actionBarSize"
+            android:background="?attr/colorPrimary"
+            app:popupTheme="@style/Theme.Androidtask.PopupOverlay" />
+
+    </com.google.android.material.appbar.AppBarLayout>
+
+    <include layout="@layout/content_main" />
+
+    <!--<com.google.android.material.floatingactionbutton.FloatingActionButton
```
Comment (daddymercury) [inline app/src/main/res/layout/app_bar_main.xml]: delte commented code

File: app/src/main/res/layout/fragment_gallery.xml [hunk 32]
```diff
@@ -0,0 +1,36 @@
...
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".ui.gallery.GalleryFragment">
+
+    <TextView
+        android:id="@+id/text_gallery"
+        android:layout_width="392dp"
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_gallery.xml]: no hard code

File: app/src/main/res/layout/fragment_gallery.xml [hunk 33]
```diff
@@ -0,0 +1,36 @@
...
+        android:layout_height="39dp"
+        android:layout_marginStart="8dp"
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_gallery.xml]: add this into [dimens](https://stackoverflow.com/questions/29128178/how-to-use-dimens-xml-in-android)

File: app/src/main/res/layout/fragment_gallery.xml [hunk 34]
```diff
@@ -0,0 +1,36 @@
...
+        android:layout_marginTop="8dp"
+        android:layout_marginEnd="8dp"
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_gallery.xml]: add this into [dimens](https://stackoverflow.com/questions/29128178/how-to-use-dimens-xml-in-android)

File: app/src/main/res/layout/fragment_gallery.xml [hunk 35]
```diff
@@ -0,0 +1,36 @@
...
+        app:layout_constraintHorizontal_bias="0.0"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintVertical_bias="0.025" />
+
+
+    <EditText
+        android:id="@+id/watchedText"
+        android:layout_width="387dp"
+        android:layout_height="39dp"
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_gallery.xml]: no hard code values

File: app/src/main/res/layout/fragment_gallery.xml [hunk 36]
```diff
@@ -0,0 +1,36 @@
...
+
+    <TextView
+        android:id="@+id/text_gallery"
+        android:layout_width="392dp"
+        android:layout_height="39dp"
+        android:layout_marginStart="8dp"
+        android:layout_marginTop="8dp"
+        android:layout_marginEnd="8dp"
+        android:textAlignment="center"
+        android:textSize="20sp"
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_gallery.xml]: add this into [dimens](https://stackoverflow.com/questions/29128178/how-to-use-dimens-xml-in-android)

File: app/src/main/res/layout/fragment_gallery.xml [hunk 37]
```diff
@@ -0,0 +1,36 @@
...
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.0"
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_gallery.xml]: delete bias

File: app/src/main/res/layout/fragment_gallery.xml [hunk 38]
```diff
@@ -0,0 +1,36 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintVertical_bias="0.025" />
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_gallery.xml]: delete bias

File: app/src/main/res/layout/fragment_home.xml [hunk 39]
```diff
@@ -0,0 +1,49 @@
...
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".ui.home.HomeFragment">
+
+    <TextView
+        android:id="@+id/text_home"
+        android:layout_width="391dp"
+        android:layout_height="36dp"
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_home.xml]: no hard code

File: app/src/main/res/layout/fragment_home.xml [hunk 40]
```diff
@@ -0,0 +1,49 @@
...
+        android:layout_marginStart="8dp"
+        android:layout_marginTop="8dp"
+        android:layout_marginEnd="8dp"
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_home.xml]: add this into [dimens](https://stackoverflow.com/questions/29128178/how-to-use-dimens-xml-in-android)

File: app/src/main/res/layout/fragment_slideshow.xml [hunk 41]
```diff
@@ -0,0 +1,36 @@
...
+    android:layout_height="match_parent"
+    tools:context=".ui.slideshow.SlideshowFragment">
+
+    <TextView
+        android:id="@+id/text_slideshow"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="8dp"
+        android:layout_marginTop="8dp"
+        android:layout_marginEnd="8dp"
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_slideshow.xml]: add this into [dimens](https://stackoverflow.com/questions/29128178/how-to-use-dimens-xml-in-android) and in the future places!

File: app/src/main/res/layout/fragment_slideshow.xml [hunk 42]
```diff
@@ -0,0 +1,36 @@
...
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="1.0"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintVertical_bias="0.011" />
+
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/recyclerView"
+        android:layout_width="304dp"
+        android:layout_height="589dp"
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_slideshow.xml]: delete hardcode here and bias in the future places

File: app/src/main/res/layout/fragment_slideshow.xml [hunk 43]
```diff
@@ -0,0 +1,36 @@
...
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.495"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_slideshow.xml]: empty line

File: app/src/main/res/layout/item.xml [hunk 44]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:layout_margin="8dp"
+    android:background="#8EAF9E">
+
+    <!--   <TextView
```
Comment (daddymercury) [inline app/src/main/res/layout/item.xml]: commented code

File: app/src/main/res/menu/main.xml [hunk 45]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+ <!--
```
Comment (daddymercury) [inline app/src/main/res/menu/main.xml]: commented?

**Comment by daddymercury** [comment]
Hey @chikebaba , nice work by the way, please fix my comments

**Review by daddymercury — CHANGES_REQUESTED** [review]

**Review by chikebaba — COMMENTED** [review]
Changes according to review (code and dimens)

File: app/src/main/java/com/example/android_task/ui/gallery/GalleryFragment.kt (hunk 12 above)
Comment (daddymercury) [reply app/src/main/java/com/example/android_task/ui/gallery/GalleryFragment.kt:41]: https://developer.android.com/reference/kotlin/androidx/core/widget/package-summary#(android.widget.TextView).doOnTextChanged(kotlin.Function4)

Comment (daddymercury) [reply app/src/main/java/com/example/android_task/ui/gallery/GalleryFragment.kt:41]: use this method here and delete Textwatcher with 3 methods

File: app/src/main/java/com/example/android_task/ui/gallery/GalleryViewModel.kt [hunk 46]
```diff
@@ -7,7 +7,7 @@ import androidx.lifecycle.ViewModel
 class GalleryViewModel : ViewModel() {
 
     private val _text = MutableLiveData<String>().apply {
-        value = ""
+        value = "This is fragment with echo editText"
```
Comment (daddymercury) [inline app/src/main/java/com/example/android_task/ui/gallery/GalleryViewModel.kt:10]: It looks like this string used in 2+ places add please this into constants

File: app/src/main/res/layout/fragment_gallery.xml [hunk 47]
```diff
@@ -8,29 +8,28 @@
 
     <TextView
         android:id="@+id/text_gallery"
-        android:layout_width="392dp"
-        android:layout_height="39dp"
-        android:layout_marginStart="8dp"
-        android:layout_marginTop="8dp"
-        android:layout_marginEnd="8dp"
+        android:layout_width="match_parent"
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_gallery.xml]: use 0 dp instead

File: app/src/main/res/layout/fragment_home.xml [hunk 48]
```diff
@@ -8,25 +8,24 @@
 
     <TextView
         android:id="@+id/text_home"
-        android:layout_width="391dp"
-        android:layout_height="36dp"
-        android:layout_marginStart="8dp"
-        android:layout_marginTop="8dp"
-        android:layout_marginEnd="8dp"
+        android:layout_width="match_parent"
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_home.xml]: please use 0dp

File: app/src/main/res/layout/fragment_home.xml [hunk 49]
```diff
@@ -8,25 +8,24 @@
...
         android:gravity="center_vertical|center_horizontal"
         android:textAlignment="center"
-        android:textSize="20sp"
+        android:textSize="@dimen/but_textSize"
         app:layout_constraintBottom_toBottomOf="parent"
         app:layout_constraintEnd_toEndOf="parent"
-        app:layout_constraintHorizontal_bias="0.0"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toTopOf="parent"
         app:layout_constraintVertical_bias="0.011" />
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_home.xml]: try to avoid using bias

File: app/src/main/res/layout/fragment_home.xml [hunk 50]
```diff
@@ -8,25 +8,24 @@
...
 
     <TextView
         android:id="@+id/textView3"
-        android:layout_width="359dp"
-        android:layout_height="49dp"
+        android:layout_width="match_parent"
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_home.xml]: please use 0 dp instead

File: app/src/main/res/layout/fragment_home.xml [hunk 51]
```diff
@@ -37,13 +36,13 @@
 
     <TextView
         android:id="@+id/textView2"
-        android:layout_width="359dp"
-        android:layout_height="49dp"
+        android:layout_width="match_parent"
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_home.xml]: please use 0 dp instead

File: app/src/main/res/layout/fragment_slideshow.xml [hunk 52]
```diff
@@ -24,13 +24,11 @@
 
     <androidx.recyclerview.widget.RecyclerView
         android:id="@+id/recyclerView"
-        android:layout_width="304dp"
-        android:layout_height="589dp"
+        android:layout_width="match_parent"
```
Comment (daddymercury) [inline app/src/main/res/layout/fragment_slideshow.xml]: please use 0 dp instead

**Review by daddymercury — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/android_task/ui/gallery/GalleryViewModel.kt (hunk 46 above)
Comment (chikebaba) [reply app/src/main/java/com/example/android_task/ui/gallery/GalleryViewModel.kt:10]: It is used only in one place - in second fragment in one textview

**Pull request merged**